src/se/vidstige/jadb/JadbConnection.java

```diff
@@ -25,6 +25,20 @@ public Transport createTransport() throws IOException {
         return new Transport(new Socket(host, port));
     }
 
+    public void connect(String ip, int port) throws IOException, JadbException {
+        Transport main = this.createTransport();
+        main.send("host:connect:" + ip + ":" + port);
+        main.verifyResponse();
+        main.close();
+    }
```

Comment:
Alright, so it restarts the daemon and starts listening to the specified port. It's not described in the [documentation](https://github.com/android/platform_system_core/blob/master/adb/SERVICES.TXT). Have you tested this works?